scripts/lint.sh

```diff
@@ -32,7 +32,9 @@ fi
 TESTS=${TESTS:-"golangci_lint license_header require_error_is_no_funcs_as_params single_import interface_compliance_nil require_no_error_inline_func"}
 
 function test_golangci_lint {
-  go install -v github.com/golangci/golangci-lint/cmd/golangci-lint@v1.56.1
+  # We use the commit hash (dc2815312726b4f374f5c69a1c183a85990917a7) to reference the specific version (v1.58.1) used in our build process.
+  # This guarantees a known-good and verifiable version. Unlike releases, commits cannot be altered or deleted after they are made, providing a more reliable reference point for our build system.
```

Comment:
Does this actually protect against anything? Google maintains protections against changing the checksum of the provided code against a given tag already... If we assumed this protection didn't exist, couldn't the maintainer of the repo just delete the old commit and tag a new commit with the hash of the old commit?

Unless I'm missing something this feels like some additional complexity that we probably don't need to maintain.

Reply:
afaik, the protection against changing the hash is only in go.mod/go.sum. Not for "random" tools being installed.
Let's hash that in private.

Reply:
I've rollback this change so that we can de-couple this one from this PR. I will keep working on validating this in the background.